internal/internal_utils.go

```diff
@@ -98,7 +100,19 @@ func chanTimeout(timeout time.Duration) func(builder *contextBuilder) {
 
 // newChannelContext - Get a rpc channel context
 func newChannelContext(ctx context.Context, options ...func(builder *contextBuilder)) (context.Context, context.CancelFunc, []yarpc.CallOption) {
-	builder := &contextBuilder{Timeout: defaultRPCTimeout}
+	rpcTimeout := defaultRPCTimeout
+	if ctx != nil {
```

Comment:
does this also apply to long polls ?

Reply:
LongPoll API calls this with context.Background without any timeout and explicitly sets the timeout of 3 minutes using options.